packages/react-google-maps-api/src/GoogleMapNext.tsx

```diff
@@ -0,0 +1,287 @@
+  zoom?: number
+  /** This event is fired when the user clicks on the map. An ApiMouseEvent with properties for the clicked location is returned unless a place icon was clicked, in which case an IconMouseEvent with a placeid is returned. IconMouseEvent and ApiMouseEvent are identical, except that IconMouseEvent has the placeid field. The event can always be treated as an ApiMouseEvent when the placeid is not important. The click event is not fired if a marker or infowindow was clicked. */
+  onClick?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the user double-clicks on the map. Note that the click event will also fire, right before this one. */
+  onDblClick?: (e: google.maps.MouseEvent) => void
+  /** This event is repeatedly fired while the user drags the map. */
+  onDrag?: () => void
+  /** This event is fired when the user stops dragging the map. */
+  onDragEnd?: () => void
+  /** This event is fired when the user starts dragging the map. */
+  onDragStart?: () => void
+  /** This event is fired when the mapTypeId property changes. */
+  onMapTypeIdChanged?: () => void
+  /** This event is fired whenever the user's mouse moves over the map container. */
+  onMouseMove?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the user's mouse exits the map container. */
+  onMouseOut?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the user's mouse enters the map container. */
+  onMouseOver?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the DOM contextmenu event is fired on the map container. */
+  onRightClick?: (e: google.maps.MouseEvent) => void
+  /** This event is fired when the visible tiles have finished loading. */
+  onTilesLoaded?: () => void
+  /** This event is fired when the viewport bounds have changed. */
+  onBoundsChanged?: () => void
+  /** This event is fired when the map center property changes. */
+  onCenterChanged?: () => void
+  /** This event is fired when the map heading property changes. */
+  onHeadingChanged?: () => void
+  /** This event is fired when the map becomes idle after panning or zooming. */
+  onIdle?: () => void
+  /** This event is fired when the projection has changed. */
+  onProjectionChanged?: () => void
+  /** This event is fired when the map size has changed. */
+  onResize?: () => void
+  /** This event is fired when the map tilt property changes. */
+  onTiltChanged?: () => void
+  /** This event is fired when the map zoom property changes. */
+  onZoomChanged?: () => void
+  /** This callback is called when the component unmounts. It is called with the map instance. */
+  onLoad?: (map: google.maps.Map) => void | Promise<void>
+  /** This callback is called when the map instance has loaded. It is called with the map instance. */
+  onUnmount?: (map: google.maps.Map) => void | Promise<void>
+}
+
+export function GoogleMap(props: GoogleMapProps): JSX.Element {
```

Comment:
I'm not sure if the GoogleMap component follows the exact same logic as the others, but when I initially tried implementing hooks the idea was creating a `useMapComponent` hook that accepts the component type, it's eventMap and updaterMap. 

Check out (not a working implementation, but the idea I think is pretty much there):
The "generic" hook:
https://github.com/JustFly1984/react-google-maps-api/blob/1b1940eef9a293c7e4743484bb75046e65bd8d5b/packages/react-google-maps-api/src/utils/use-map-component.ts

And a component that uses it:
https://github.com/JustFly1984/react-google-maps-api/blob/1b1940eef9a293c7e4743484bb75046e65bd8d5b/packages/react-google-maps-api/src/components/drawing/Rectangle.tsx

Reply:
@uriklar curious, any reason this example omits the onLoad prop ? it's not due to any limitation is it?